src/Maestro/tests/Maestro.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -325,8 +423,24 @@ await using (await PushGitBranchAsync("origin", targetBranch))
                                 return;
                             }
 
+                            if (isFeedTest)
+                            {
+                                await CheckNonBatchedAzDoPullRequest(sourceRepoName, targetRepoName, targetBranch, expectedAzDoFeedFlowDependencies, reposFolder.Directory, isCompleted: false, isUpdated: false, expectedFeeds: expectedFeeds, notExpectedFeeds: notExpectedFeeds);
+                            }
+
                             // The remaining non-batched tests continue to make sure that updating works as expected
-                            await CheckNonBatchedGitHubPullRequest(sourceRepoName, targetRepoName, targetBranch, expectedDependenciesSource1, reposFolder.Directory, false);
+                            if (isAzDoTest)
```

Comment:
It feels like this class is somehow three classes masquerading as one.  A _lot_ of the code has this strange "do one of three things" blocks.

That seems to mean this class is doing too much.  Is there some common setup and the different checks? And there could be a "EndToEndSetup" class that could contain that?  Is it that there are a lot of common check but some are different, and it needs to be an EndToEndVerification class, and the specific tests can do some additional tests?

This class is so large that I can't tell what it's purpose is other than "everything".

Reply:
It's tricky to split it out into setup vs running or even into "do this here and do that in the test case" because a lot of things are scoped for disposal in a way that would break if things were split up too much.

Generally speaking, the deal is that the logic is the same for most of the tests, but they all have different inputs and some of them continue with additional checks after that identical logic. I'm splitting it up into GitHub vs AzDo and then within that into PRs that update and PRs that don't. This duplicates a chunk of code but hopefully makes it easier to read and follow.